Makefile

```diff
@@ -285,18 +289,69 @@ ifdef MACOS_CODESIGN_IDENT
 	scripts/sign-macos-pkg.sh $(STATUS_CLIENT_DMG) $(MACOS_CODESIGN_IDENT)
 endif
 
-STATUS_CLIENT_EXE ?= pkg/Status.exe
-
-# not implemented yet
-# $(STATUS_CLIENT_EXE): nim_status_client
+NIM_WINDOWS_PREBUILT_DLLS ?= tmp/windows/tools/pcre.dll
+
+$(NIM_WINDOWS_PREBUILT_DLLS):
+	echo -e "\e[92mFetching:\e[39m prebuilt DLLs from nim-lang.org"
+	rm -rf tmp/windows
+	mkdir -p tmp/windows/tools
+	cd tmp/windows/tools && \
+	wget https://nim-lang.org/download/dlls.zip && \
+	unzip dlls.zip
+
+nim_windows_launcher: | deps
```

Comment:
I received this error when running this on my machine:
```
Hint:  [Link]
/usr/bin/ld: cannot find -lwinpthread
/usr/bin/ld: nimcache/debug/nim_windows_launcher/stdlib_dynlib.nim.c.o: in function `loadLib__Yq5XYz2ycX5V5B9bUM4Uyiw':
/home/richard/status/nim-status-client/vendor/nimbus-build-system/vendor/Nim/lib/pure/dynlib.nim:125: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: 'g++   -o /home/richard/status/nim-status-client/bin/nim_windows_launcher  nimcache/debug/nim_windows_launcher/stdlib_assertions.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_locks.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_sharedlist.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_widestrs.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_system.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_pathnorm.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_times.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_os.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_dynlib.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_winlean.nim.c.o nimcache/debug/nim_windows_launcher/@mnim_windows_launcher.nim.c.o  -pthread -lm -lrt  -lcrypto -lssl -Wl,-as-needed -static-libgcc -Wl,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive  -ldl'
make: *** [Makefile:303: nim_windows_launcher] Error 1
```
Should i use only a windows machine for this target?

Reply:
The Windows building and packaging logic in the Makefile assumes it's being run on a Windows machine, with `mingw32-make.exe` in a Git Bash shell.

On my Windows machine, `libwinpthread-1.dll` resolves to `/c/Qt/Tools/mingw730_64/bin/libwinpthread-1.dll`.

Reply:
> the msvc toolchain is involved in building `DOtherSide.dll`

Now I'm wondering why your build didn't fail earlier. I guess that if you tried it on Linux then DOtherSide was built the usual way there. We don't have explicit `build-macos|linux|windows` targets so maybe the `pkg-macos|linux|windows` targets should exit with an error message if the OS you're building on isn't a match for the OS the build is targeting.

Reply:
That would be nice, because otherwise displaying these build errors could make someone think there's an issue in the build system